scripts-dev/check_locked_deps_have_sdists.py

```diff
@@ -24,9 +24,16 @@ def main() -> None:
     with open(lockfile_path, "rb") as lockfile:
         lockfile_content = tomli.load(lockfile)
 
-    packages_to_assets: Dict[str, List[Dict[str, str]]] = lockfile_content["metadata"][
-        "files"
-    ]
+    packages_to_assets: Dict[str, List[Dict[str, str]]]
+    # There seem to be two different formats for storing the list of files per package.
+    if lockfile_content.get("metadata", {}).get("files") is not None:
+        # either [metadata.files]
+        packages_to_assets = lockfile_content["metadata"]["files"]
+    else:
+        # or a `files` inline table in each [[package]]
+        packages_to_assets = {
+            package["name"]: package["files"] for package in lockfile_content["package"]
+        }
```

Comment:
Seems to be part of the poetry 1.3 lockfile format, see https://github.com/python-poetry/poetry/pull/6393

Reply:
Note that I'm about to require the new lockfile format in https://github.com/matrix-org/sytest/pull/1327